2-projects/1-todo-list/script.js

```diff
@@ -9,17 +38,40 @@ let todos = [
   { task: "Wash the dishes", completed: false },
   { task: "Do the shopping", completed: false },
 ];
+console.log(todos);
+let form = document.getElementById("todo-form");
+form.addEventListener("submit", addNewTodo);
 
-populateTodoList(todos);
-
-// This function will take the value of the input field and add it as a new todo to the bottom of the todo list. These new todos will need the completed and delete buttons adding like normal.
-function addNewTodo(event) {
-  // The code below prevents the page from refreshing when we click the 'Add Todo' button.
-  event.preventDefault();
-  // Write your code here... and remember to reset the input field to be blank after creating a todo!
+function addNewTodo(e) {
+  e.preventDefault();
+  if (!input.value) {
+    return;
+  } else {
+    let obj = Object.assign({
+      task: input.value,
+      completed: window.getComputedStyle(e.target).textDecoration
+        ? true
+        : false,
+    });
+    todos.push(obj);
+  }
+  populateTodoList(input.value);
+  input.value = "";
 }
 
+// function removeTodo(todo) {
+//   // var todos = getTodos()
+//   var newTodos = todos.filter(function (el) {
+//     return el !== todo;
+//   });
+//   saveTodos(newTodos);
+
+//   // saveTodos(getTodos().filter(el => el !== todo))
+// }
 // Advanced challenge: Write a fucntion that checks the todos in the todo list and deletes the completed ones (we can check which ones are completed by seeing if they have the line-through styling applied or not).
+const del = document.getElementById("del");
+
 function deleteAllCompletedTodos() {
-  // Write your code here...
+  todos.forEach((element) => {});
 }
+del.addEventListener("click", deleteAllCompletedTodos);
```

Comment:
This all works well. An alternative approach would be to add, delete or update items in the underlying todos list when an add, delete or complete/uncomplete event is called, and then call the populateTodoList with your updated todos.